stablehlo/dialect/TypeInference.cpp

```diff
@@ -3272,6 +3323,13 @@ LogicalResult verifyBroadcastInDimOp(std::optional<Location> location,
     return success();
   }
 
+  // broadcast_in_dim_c1
+  // part of broadcast_in_dim_c1 already verified by
+  // Trait HLO_CompatibleOperandsAndResultElementType
+  if (failed(sameQuantScaleZeroPoint(location, operand.getType(),
```

Comment:
I note here that `broadcast_in_dim` should not use the trait introduced at #2112. At the constrainsts on scales/zps for per-tensor and per-axis are different.